Practical Dependency InversionTests/MainThreadDecoratorTests.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  MainThreadTests.swift
+//  Practical Dependency InversionTests
+//
+//  Created by 游宗諭 on 2020/7/14.
+//  Copyright © 2020 Hanyu. All rights reserved.
+//
+
+import XCTest
+@testable import Practical_Dependency_Inversion
+
+class MainThreadDecoratorTests: XCTestCase {
+    
+    
+    func testCanCheckNotOnMainQueue() throws {
+        let testQueue = DispatchQueue(label: #function)
+        let await = XCTestExpectation(description: #function)
+        var result:Bool?
+        testQueue.async {
+            checkQueue(queue: .main, result: &result)
+            await.fulfill()
+        }
+        wait(for: [await], timeout: 1)
+        let returned = try XCTUnwrap(result)
+        XCTAssertFalse(returned)
+    }
+    func testCanCheckIsOnMainQueue() throws {
+        let testQueue = DispatchQueue(label: #function)
+        let await = XCTestExpectation(description: #function)
+        var result:Bool?
+        testQueue.async {
```

Comment:
[Q]
Is this necessary?

Reply:
The main purpose of this test is to test some statement should run on the main queue. To do that, we need to make sure that the current queue must not main queue, that just reminds me that I did not test for it. I will do some modifications.

Reply:
Added the assertion before redirecting to the main queue, make sure there are different.

Reply:
Ah! Now that makes perfect sense. Looks great!

Reply:
Thanks a lot!